core/envelope-sim/src/main/kotlin/fr/sncf/osrd/envelope_sim/EnvelopeSimContext.kt

```diff
@@ -28,19 +29,42 @@ constructor(
          * *after* the end of the path (or right at the end).
          */
         val dangerPointOffsets: List<Offset<TravelledPath>>,
+        /**
+         * List of ETCS-block detector offsets for every block on the path. Starts at the start of
+         * the path, can end after the end of the path.
+         */
+        val detectorOffsets: List<Offset<TravelledPath>>
     ) {
         /**
-         * Returns the SVL location: next buffer stop or switch, whichever is closest. If there is
-         * any.
+         * Returns the next danger point location: next buffer stop or switch, whichever is closest.
+         * If there is any.
+         */
+        fun getDangerPoint(offset: Offset<TravelledPath>): Offset<TravelledPath>? {
+            return dangerPointOffsets.firstOrNull { it >= offset }
+        }
+
+        /**
+         * On a closed signal of for an interlocking signal, a danger point is always located less
+         * than 200m away. This method returns the next danger point location if it is less than
+         * 200m away. If there isn't any, we're probably missing a danger point in the
+         * infrastructure, hence we'll be conservative and place the danger point on the closed
+         * signal.
```

Comment:
Not sure it's the best phrasing, but on top and bottom parts of the comment we're currently missing the fact that for a closed-signal stop, EoA is at the stop's location, not the signal:
```suggestion
         * infrastructure, hence we'll be conservative and place the danger point on the EoA (stop location or signal)
```